### PR TITLE
SSCSCI-879: Validate DirectionDueDate only when explicitly entered

### PR DESCRIPTION
**Pull request by hashimalisolirius**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/SSCSCI-879

### Change description ###
Add new callback for adjournCaseDirectionsDueDateDaysOffset
Separate out write adjournment event fields into CaseEventsToFields-writeAdjournmentNotice.json

### Checklist

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

**Review by KatKovacs1 — APPROVED** [review]

**Review by NHAL22 — APPROVED** [review]

File: benefit/data/sheets/CaseEventToFields/CaseEventsToFields-writeAdjournmentNotice.json [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+    "LiveFrom": "01/01/2018",
+    "PageDisplayOrder": 6,
+    "PageFieldDisplayOrder": 1,
+    "PageID": "areDirectionsBeingMadeToParties",
+    "PageLabel": "Are you making directions to the parties?",
+    "PageShowCondition": "adjournCaseGenerateNotice = \"No\" OR adjournCaseCanCaseBeListedRightAway = \"No\"",
+    "ShowSummaryChangeOption": "Y"
+  },
+  {
+    "CallBackURLMidEvent": "${CCD_DEF_TRIBUNALS_API_URL}/ccdMidEventAdjournDirectionDueDate",
```
Comment (benouaer) [inline benefit/data/sheets/CaseEventToFields/CaseEventsToFields-writeAdjournmentNotice.json:78]: can we follow the same naming convention as they've done for the other midEvent
${CCD_DEF_TRIBUNALS_API_URL}/ccdMidEventAdjournCasePopulateVenueDropdown

File: benefit/data/sheets/CaseEventToFields/CaseEventsToFields-writeAdjournmentNotice.json [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+    "LiveFrom": "01/01/2018",
+    "PageDisplayOrder": 14,
+    "PageFieldDisplayOrder": 1,
+    "PageID": "reasons",
+    "PageLabel": "Reasons for adjournment",
+    "PageShowCondition": "adjournCaseGenerateNotice = \"Yes\"",
+    "ShowSummaryChangeOption": "Y"
+  },
+  {
+    "CallBackURLMidEvent": "${CCD_DEF_TRIBUNALS_API_URL}/ccdMidEventPreviewAdjournCase",
```
Comment (benouaer) [inline benefit/data/sheets/CaseEventToFields/CaseEventsToFields-writeAdjournmentNotice.json:306]: can we follow the same naming convention as they've done for the other midEvent
${CCD_DEF_TRIBUNALS_API_URL}/ccdMidEventAdjournCasePopulateVenueDropdown

File: benefit/data/sheets/CaseEventToFields/CaseEventsToFields-writeAdjournmentNotice.json [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+    "LiveFrom": "01/01/2018",
+    "PageDisplayOrder": 12,
+    "PageFieldDisplayOrder": 2,
+    "PageID": "interpreterRequired",
+    "PageLabel": "Language interpreter",
+    "ShowSummaryChangeOption": "Y",
+    "RetainHiddenValue": "Yes"
+  },
+  {
+    "CallBackURLMidEvent": "${CCD_DEF_TRIBUNALS_API_URL}/ccdMidEvent",
```
Comment (benouaer) [inline benefit/data/sheets/CaseEventToFields/CaseEventsToFields-writeAdjournmentNotice.json:227]: this is the midEvent that we want to create a specific handler, isn't it?
if so, we should follow the same naming convention as they've done for the other midEvent
${CCD_DEF_TRIBUNALS_API_URL}/ccdMidEventAdjournCasePopulateVenueDropdown

File: benefit/data/sheets/CaseEventToFields/CaseEventsToFields-writeAdjournmentNotice.json [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+    "LiveFrom": "01/01/2018",
+    "PageDisplayOrder": 15,
+    "PageFieldDisplayOrder": 1,
+    "PageID": "additionalDirections",
+    "PageLabel": "Additional directions",
+    "PageShowCondition": "adjournCaseGenerateNotice = \"Yes\"",
+    "ShowSummaryChangeOption": "Y"
+  },
+  {
+    "CallBackURLMidEvent": "${CCD_DEF_TRIBUNALS_API_URL}/ccdMidEvent",
```
Comment (benouaer) [inline benefit/data/sheets/CaseEventToFields/CaseEventsToFields-writeAdjournmentNotice.json:320]: what's the use of this midEvent? not sure it does anything

**Review by NHAL22 — DISMISSED** [review]

**Review by benouaer — DISMISSED** [review]

**Review by benouaer — APPROVED** [review]

**Review by nitinprabhuhmcts — DISMISSED** [review]

**Review by KatKovacs1 — APPROVED** [review]

**Review by nitinprabhuhmcts — APPROVED** [review]

**Pull request merged**